pkg/danm/danm.go

```diff
@@ -183,72 +188,94 @@ func extractConnections(args *cniArgs) error {
   return nil
 }
 
-func setupNetworking(args *cniArgs) (*current.Result, error) {
-  syncher := syncher.NewSyncher(len(args.interfaces))
-  for nicId, nicParams := range args.interfaces {
-    nicParams.DefaultIfaceName = "eth" + strconv.Itoa(nicId)
-    go createInterface(syncher, nicParams, args)
+func getAllocatedDevices(args *cniArgs, devicePool string)(*[]string, error){
+  checkpoint, err := checkpoint_utils.GetCheckpoint()
+  if err != nil {
+    return nil, errors.New("failed to instantiate checkpoint object due to:" + err.Error())
   }
-  err := syncher.GetAggregatedResult()
-  return syncher.MergeCniResults(), err
+  resourceMap, err := checkpoint.GetComputeDeviceMap(string(args.podUid))
+  if err != nil || len(resourceMap) == 0 {
+    return nil, errors.New("failed to retrieve Pod info from checkpoint object due to:" + err.Error())
+  }
+  if _, ok := resourceMap[devicePool]; !ok {
+    return nil, errors.New("failed to retrieve resources of DevicePool")
+  }
+  return &resourceMap[devicePool].DeviceIDs, nil
```

Comment:
a slice is actually a pointer anyway, so it is kind of strange to dereference it

what I meant is passing the map by a pointer

I think the most elegant way would be creating some named types, like Devices []string, and DeviceMap map[string]Devices
then pass *DeviceMap to popDevice as input

would increase readability IMO by a big margin, but I leave it up to you

Reply:
If you would accept, I would leave it as it is

Reply:
yeah no worries, maybe I will re-factor on my own when I'm bored